apps/prairielearn/src/lib/client/page-context.test.ts

```diff
@@ -0,0 +1,143 @@
+import { describe, expect, it } from 'vitest';
+
+import {
+  type StaffCourseInstanceContext,
+  type StudentCourseInstanceContext,
+  getCourseInstanceContext,
+  getPageContext,
+} from './page-context.js';
+
+describe('getPageContext', () => {
+  it('strips extra fields from the data', () => {
+    const mockData = {
+      authz_data: {
+        has_course_instance_permission_edit: true,
+        has_course_instance_permission_view: true,
+        has_course_permission_own: true,
+        user: { name: 'Test User', uid: 'test@illinois.edu' },
+        mode: 'edit',
+      },
+      urlPrefix: '/pl/course/1/course_instance/1',
+      access_as_administrator: false,
+      news_item_notification_count: 0,
+      authn_is_administrator: false,
+      authn_user: { name: 'Test User', uid: 'test@illinois.edu' },
+      viewType: 'instructor',
+      extraField: 'this should be stripped',
+      anotherExtraField: 123,
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { extraField, anotherExtraField, ...expected } = mockData;
+
+    const result = getPageContext(mockData);
+
+    expect(result).toEqual(expected);
+  });
+
+  it('throws error when required fields are missing', () => {
+    const invalidData = {
+      // Missing most fields
+      authz_data: {
+        has_course_instance_permission_edit: true,
+      },
+    };
+
+    expect(() => getPageContext(invalidData)).toThrow();
+  });
+});
+
+describe('getCourseInstanceContext', () => {
+  const mockStudentData: StudentCourseInstanceContext = {
+    course_instance: {
+      assessments_group_by: 'Set',
+      course_id: '1',
+      deleted_at: null,
+      display_timezone: 'America/Chicago',
+      hide_in_enroll_page: false,
+      id: '1',
+      json_comment: null,
+      long_name: 'Example Student Course Instance',
+      share_source_publicly: true,
```

Comment:
This probably shouldn't be visible to students.

Reply:
Updated in a9e5a71c2310a82015c4a2e1633d3681d1391547.